Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -229,6 +214,14 @@ function New-AvsLDAPIdentitySource {
         [string]
         $GroupName
     )
+
+    if (-not ($PrimaryUrl -match 'ldap:.*((389)|(636)|(3268)(3269))')) {
+        Write-Error "$PrimaryUrl is invalid. Ensure the port number is 389, 636, 3268, or 3269 and that the url begins with ldap:" -ErrorAction Stop
+    }
+    if ($PSBoundParameters.ContainsKey('SecondaryUrl') -and (-not ($SecondaryUrl -match 'ldap:.*((389)|(636)|(3268)(3269))'))) {
```

Comment:
How strict should this regex be? Currently it considers valid things like 'ldap:389' and 'oopsldap:avs:389oops'.
Maybe '^(ldap:).+((:389)|(:636)|(:3268)|(:3269))$' to ensure the url starts and ends correctly with at least 1 character between?

Reply:
More strict is always the better.